src/daft-csv/src/read.rs

```diff
@@ -145,12 +190,42 @@ async fn read_csv_single_into_table(
     io_stats: Option<IOStatsRef>,
     max_chunks_in_flight: Option<usize>,
 ) -> DaftResult<Table> {
+    let predicate = convert_options
+        .as_ref()
+        .and_then(|opts| opts.predicate.clone());
+
+    let limit = convert_options.as_ref().and_then(|opts| opts.limit);
+
+    let mut remaining_rows = limit.map(|limit| limit as i64);
```

Comment:
Nit: Might be more readable if the `remaining_rows` definition was moved to be closer to the `.try_take_while()` call?